### PR TITLE
feat: Update trailblazer check for multiple periodic donations

### PR DESCRIPTION
**Pull request by howar31**
# Issue
All periodic donation records of one user should be included while evaluating the role.

# Notice

# Dependency

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205131588158234

### CODE REVIEW

File: storage/user.go [hunk 1]
```diff
@@ -262,3 +262,31 @@ func (gs *GormStorage) AssignRoleToUser(user models.User, roleKey string) error
...
+func (gs *GormStorage) IsTrailblazer(email string) (bool, error) {
+	var result struct {
+		Sum int
+	}
+
+	query := gs.db.Model(&models.User{}).
+		Joins("JOIN periodic_donations p ON users.id = p.user_id").
+		Joins("JOIN pay_by_card_token_donations pd ON p.id = pd.periodic_id").
+		Where("pd.status = ?", "paid").
+		Where("p.status = ?", "paid").
```
Comment (nickhsine) [inline storage/user.go:276]: Question 🙋 :
目前「會員身份的調整」是以近兩個月的捐款紀錄來做判斷，要連續捐款失敗兩個月以上，我們才會暫停該筆捐款。
如果我們只判斷 `status = paid` 的狀況，會不會提早將會員降等？

File: storage/user.go [hunk 2]
```diff
@@ -262,3 +262,31 @@ func (gs *GormStorage) AssignRoleToUser(user models.User, roleKey string) error
...
+// Evaluate given user's role by the periodic donation records
+func (gs *GormStorage) IsTrailblazer(email string) (bool, error) {
+	var result struct {
+		Sum int
+	}
+
+	query := gs.db.Model(&models.User{}).
+		Joins("JOIN periodic_donations p ON users.id = p.user_id").
+		Joins("JOIN pay_by_card_token_donations pd ON p.id = pd.periodic_id").
+		Where("pd.status = ?", "paid").
```
Comment (Aylie-Chou) [inline storage/user.go:275]: Question: why do we need to check `pd` paid status ?

Comment (howar31) [reply storage/user.go:275]: In DB, `periodic_donations.status` has:
- paid
- fail
- stopped
- invalid

and `pay_by_card_token_donations.status` has:
- paid
- failed
- refunded

According to spec `因取消任一訂單（訂單狀況為 stopped)而不符合開創者身分時，即時降等成行動者，並寄送系統信件`, the `stopped` condition is in `periodic_donations`

File: storage/user.go (hunk 1 above)
Comment (howar31) [reply storage/user.go:276]: fixed

Comment (nickhsine) [reply storage/user.go:276]: 想問問，我們是不是可以透過 `periodic_donations.last_success_at` 來協助判斷呢？

File: storage/user.go (hunk 2 above)
Comment (Aylie-Chou) [reply storage/user.go:275]: I think the spec might have changed.
Please check the latest flow with @nickhsine 🙏

File: storage/user.go (hunk 1 above)
Comment (howar31) [reply storage/user.go:276]: updated

**Comment by nickhsine** [comment]
LGTM 👍

**Review by Aylie-Chou — APPROVED** [review]

**Pull request merged**